cmd/gen-go-sample/main.go

```diff
@@ -332,7 +333,15 @@ func (g *generator) genSample(ifaceName string, methConf GAPICMethod, regTag str
 	if len(argNames) > 0 {
 		var sb strings.Builder
 		for i, n := range argNames {
-			fmt.Fprintf(&sb, ", %s %s", n, pbinfo.GoTypeForPrim[argTrees[i].typ.prim])
+			if e, ok := argTrees[i].typ.desc.(*descriptor.EnumDescriptorProto); ok {
+				en, err := enumType(g.descInfo, e, g)
+				if err != nil {
+					return err
+				}
+				fmt.Fprintf(&sb, ", %s %s", n, en)
+			} else {
+				fmt.Fprintf(&sb, ", %s %s", n, pbinfo.GoTypeForPrim[argTrees[i].typ.prim])
```

Comment:
Can we make this the default?
```go
en := pbinfo.GoTypeForPrim[argTrees[i].typ.prim]
if e, ok := argTrees[i].typ.desc.(*descriptor.EnumDescriptorProto); ok {
  en, err = enumType(g.descInfo, e, g)
  if err != nil {
    return err
  }
}
fmt.Fprintf(&sb, ", %s %s", n, en)
```

Might have to declare an `err` var but this code structure flows better.

Reply:
Done